pkg/app/piped/eventwatcher/eventwatcher.go

```diff
@@ -184,23 +187,87 @@ func (w *watcher) run(ctx context.Context, repo git.Repo, repoCfg config.PipedRe
 				}
 				continue
 			}
-			cfg, err := config.LoadEventWatcher(repo.GetPath(), includedCfgs, excludedCfgs)
-			if errors.Is(err, config.ErrNotFound) {
-				w.logger.Info("configuration file for Event Watcher not found",
+			headCommit, err := repo.GetLatestCommit(ctx)
+			if err != nil {
+				w.logger.Error("failed to get latest commit",
 					zap.String("repo-id", repoCfg.RepoID),
 					zap.Error(err),
 				)
 				continue
 			}
-			if err != nil {
+			// Check whether the config file exists in .pipe/ or not and updates values if it exists.
+			// NOTE: This was deprecated and will be deleted in the future.
+			cfg, err := config.LoadEventWatcher(repo.GetPath(), includedCfgs, excludedCfgs)
+			if !errors.Is(err, config.ErrNotFound) && err != nil {
 				w.logger.Error("failed to load configuration file for Event Watcher",
 					zap.String("repo-id", repoCfg.RepoID),
 					zap.Error(err),
 				)
 				continue
 			}
-			if err := w.updateValues(ctx, repo, repoCfg.RepoID, cfg.Events, commitMsg); err != nil {
-				w.logger.Error("failed to update the values",
+			if errors.Is(err, config.ErrNotFound) {
+				w.logger.Info("there was no config file for Event Watcher in .pipe directory",
+					zap.String("repo-id", repoCfg.RepoID),
+					zap.Error(err),
+				)
+			} else {
+				if err := w.updateValues(ctx, repo, repoCfg.RepoID, cfg.Events, commitMsg); err != nil {
+					w.logger.Error("failed to update the values",
+						zap.String("repo-id", repoCfg.RepoID),
+						zap.Error(err),
+					)
+				}
+			}
+			// If event watcher config exist in the application config file, they are handled.
+			resp, err := w.apiClient.ListApplications(ctx, &pipedservice.ListApplicationsRequest{})
+			if err != nil {
+				w.logger.Error("failed to list registered application", zap.Error(err))
+				continue
+			}
+			cfgs := make([]config.EventWatcherConfig, 0, len(resp.Applications))
+			for _, app := range resp.Applications {
+				if app.GitPath.Repo.Id != repoCfg.RepoID {
+					continue
+				}
+				// Return a last scanned application if there is no new commit pushed from last scanned time for this application.
+				if v, ok := w.lastScannedConfig.Load(app.Id); ok {
+					c := v.(eventWatcherCache)
+					if c.HeadCommit == headCommit.Hash && c.Configs == nil {
+						continue
+					}
+					if c.HeadCommit == headCommit.Hash {
+						cfgs = append(cfgs, c.Configs...)
+						continue
+					}
```

Comment:
Maybe I got it. But to be simple, it can be.

```suggestion
					if c.HeadCommit == headCommit.Hash {
						cfgs = append(cfgs, c.Configs...)
						continue
					}
```

Reply:
Right, Thank you.